src/components/confirm/confirm.test.js

```diff
@@ -0,0 +1,388 @@
+
+        confirmButton().should("have.css", "background-color", backcolor);
+      }
+    );
+
+    it("should check clicking cancel button closes dialog", () => {
+      CypressMountWithProviders(<ConfirmComponent />);
+
+      cancelButton().click();
+      dialogPreview().should("not.exist");
+    });
+
+    it("should check clicking confirm button closes dialog", () => {
+      CypressMountWithProviders(<ConfirmComponent />);
+
+      confirmButton().click();
+
+      dialogPreview().should("not.exist");
+    });
+
+    it("should check clicking cancel button does not close dialog when cancel button is disabled", () => {
+      CypressMountWithProviders(<ConfirmComponent disableCancel />);
+
+      cancelButton().click({ force: true });
+
+      dialogPreview().should("be.visible");
+    });
+
+    it("should check clicking confirm button does not close dialog when confirm button is disabled", () => {
+      CypressMountWithProviders(<ConfirmComponent disableConfirm />);
+
+      confirmButton().click({ force: true });
+
+      dialogPreview().should("be.visible");
+    });
+
+    it("should check clicking cancel button does not close dialog when cancel button is disabled", () => {
+      CypressMountWithProviders(<ConfirmComponent disableEscKey />);
+
+      dialogPreview().trigger("keydown", keyCode("Esc"));
+
+      dialogPreview().should("be.visible");
+    });
+
+    it("should call click action when cancel button is clicked", () => {
+      const callback = cy.stub();
+
+      CypressMountWithProviders(<ConfirmComponent onCancel={callback} />);
+
+      cancelButton()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call click action when confirm button is clicked", () => {
+      const callback = cy.stub();
+
+      CypressMountWithProviders(<ConfirmComponent onConfirm={callback} />);
+
+      confirmButton()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+
+      it("should render Confirm with aria-label prop", () => {
+        CypressMountWithProviders(
+          <ConfirmComponent aria-label="cypress-aria" />
+        );
+
+        dialogPreview().should("have.attr", "aria-label", "cypress-aria");
+      });
+    });
+
+    it("should check confirm button has save icon", () => {
+      CypressMountWithProviders(
+        <ConfirmComponent confirmButtonIconType="Save" />
+      );
+
+      getDataElementByValue("Save").should("have.attr", "type", "Save");
+    });
+
+    it("should check cancel button has bin icon", () => {
+      CypressMountWithProviders(
+        <ConfirmComponent cancelButtonIconType="bin" />
+      );
+
+      getDataElementByValue("bin").should("have.attr", "type", "bin");
+    });
+
+    it("should check confirm dialog is not open", () => {
```

Comment:
why do we need this test? 

Why also do we need to use this component? I mean we can use the `<ConfirmComponent/>` and close it and then check